sphinx/directives/patches.py

```diff
@@ -105,13 +106,71 @@ def make_title(self):
         return title, message
 
 
+class MathDirective(SphinxDirective):
+
+    has_content = True
+    required_arguments = 0
+    optional_arguments = 1
+    final_argument_whitespace = True
+    option_spec = {
+        'label': directives.unchanged,
+        'name': directives.unchanged,
+        'nowrap': directives.flag,
+    }
+
+    def run(self):
+        # type: () -> List[nodes.Node]
+        latex = '\n'.join(self.content)
+        if self.arguments and self.arguments[0]:
+            latex = self.arguments[0] + '\n\n' + latex
+        node = nodes.math_block(latex, latex,
+                                docname=self.state.document.settings.env.docname,
+                                number=self.options.get('name'),
+                                label=self.options.get('label'),
+                                nowrap='nowrap' in self.options)
+        ret = [node]
+        set_source_info(self, node)
+        if hasattr(self, 'src'):
```

Comment:
Where this `src` attribute comes from? I don't know about this line. Simply copied from original.

Reply:
This is for docutils 0.8 or older. Directive.src instance attribute has been deleted at: https://github.com/docutils-mirror/docutils/commit/5d61d6ba8bcf3a018de46b93b5cf518080c04c33#diff-6d19a4ff14e843a848631efc1195411f
I think we can remove this fallback code.

Nevertheless, docstring of `docutils.parser.rst.Directive` still have the description of `src` (and `srcline`) attribute.
https://github.com/docutils-mirror/docutils/blob/e88c5fb08d5cdfa8b4ac1020dd6f7177778d5990/docutils/parsers/rst/__init__.py#L252-L256

relates: https://github.com/sphinx-doc/sphinx/issues/1685

Reply:
done. see #4990